[CI] TSDBSyntheticIdPostingsFormatTests testTerms failing

**Build Scans:**
- [elasticsearch-periodic #11849 / release-tests](https://gradle-enterprise.elastic.co/s/7aoc3tsa7mjjc)
- [elasticsearch-periodic #11849 / release-tests](https://gradle-enterprise.elastic.co/s/am5llqsfprzwo)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.index.codec.tsdb.TSDBSyntheticIdPostingsFormatTests.testTerms" -Dtests.seed=EF77A793EDF2481D -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dlicense.key=x-pack/license-tools/src/test/resources/public.key -Dtests.locale=ar-IQ -Dtests.timezone=Australia/Perth -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.index.codec.tsdb.TSDBSyntheticIdPostingsFormatTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testTerms'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: "OaUTN2Zm2-PaCx_vY5I6CrMLIX___mPLjmldD3RcXQ"
     but: was "XVx0D3ALZdzYxoitAAABnDRxlqI"
```

**Issue Reasons:**
- [main] 2 consecutive failures in step release-tests
- [main] 2 failures in test testTerms (5.0% fail rate in 40 executions)
- [main] 2 failures in step release-tests (100.0% fail rate in 2 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/server/src/test/java/org/elasticsearch/index/codec/tsdb/TSDBSyntheticIdPostingsFormatTests.java
+++ b/server/src/test/java/org/elasticsearch/index/codec/tsdb/TSDBSyntheticIdPostingsFormatTests.java
@@ -75,6 +75,7 @@ public class TSDBSyntheticIdPostingsFormatTests extends ESTestCase {
     private record Doc(long timestamp, String hostName, String metricField, Integer metricValue, int version, int routing) {}
 
     public void testTerms() throws IOException {
+        assumeTrue("Test should only run with feature flag", IndexSettings.TSDB_SYNTHETIC_ID_FEATURE_FLAG);
         runTest((writer, parser) -> {
 
             final var now = Instant.now();
@@ -182,6 +183,7 @@ public class TSDBSyntheticIdPostingsFormatTests extends ESTestCase {
     }
 
     public void testSeek() throws IOException {
+        assumeTrue("Test should only run with feature flag", IndexSettings.TSDB_SYNTHETIC_ID_FEATURE_FLAG);
         final int routing = randomNonNegativeInt();
         final int maxHosts = randomIntBetween(1, 25);
 

